sdk/src/tx/whileValidTxSender.ts

```diff
@@ -122,17 +122,18 @@ export class WhileValidTxSender extends BaseTxSender {
 		extraConfirmationOptions?: ExtraConfirmationOptions
 	): Promise<TxSigAndSlot> {
 		const latestBlockhash = await this.connection.getLatestBlockhash();
-		tx.message.recentBlockhash = latestBlockhash.blockhash;
```

Comment:
Explanation : changing the blockhash here was making sig verification fail for pre-signed transactions